src/reducers/ui.js

```diff
@@ -12,32 +12,26 @@ const { prefs } = require("../utils/prefs");
 import type { Action, panelPositionType } from "../actions/types";
 import type { Record } from "../utils/makeRecord";
 
-type searchVisibilityType = {
-    project: boolean,
-    document: boolean
-};
-
 export type UIState = {
-  searchVisibility: Record<searchVisibilityType>,
+  fileSearchOn: boolean,
+  projectSearchOn: boolean,
   shownSource: string,
   startPanelCollapsed: boolean,
   endPanelCollapsed: boolean,
 };
 
 const State = makeRecord(({
-  searchVisibility: makeRecord({
-    project: false,
-    document: false
-  })(),
+  fileSearchOn: false,
+  projectSearchOn: false,
   shownSource: "",
   startPanelCollapsed: prefs.startPanelCollapsed,
   endPanelCollapsed: prefs.endPanelCollapsed
 } : UIState));
 
 function update(state = State(), action: Action): Record<UIState> {
   switch (action.type) {
-    case constants.SET_FILE_SEARCH: {
-      return state.setIn(["searchVisibility", action.field], action.value);
+    case constants.TOGGLE_SEARCH: {
+      return state.set(action.field, action.value);
```

Comment:
hmm - this seems a bit too clever because it requires the redux field to leak into the API.

It is probably simpler to have two actions types `toggleFileSearch` and `toggleProjectSearch`